src/PayU/PaymentsApi/PaymentsV4/Services/ResponseParser.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+
+namespace PayU\PaymentsApi\PaymentsV4\Services;
+
+use PayU\PaymentsApi\PaymentsV4\Entities\Response\AuthorizationResponse;
+use PayU\PaymentsApi\PaymentsV4\Exceptions\AuthorizationResponseException;
+
+class ResponseParser
+{
+    /** @var AluResponseMapper */
+    private $aluResponseMapper;
+
+    public function __construct()
+    {
+        $this->aluResponseMapper = new AluResponseMapper();
+    }
+
+    /**
+     * @param string $jsonResponse
+     * @return AuthorizationResponse
+     * @throws AuthorizationResponseException
+     */
+    public function parseJsonResponse($jsonResponse)
+    {
+        $responseArray = json_decode($jsonResponse, true);
+
+        if (is_array($responseArray)) {
+            $responseArray = $this->createArray($responseArray);
```

Comment:
Could call directly $this->aluResponseMapper->processResponse

Reply:
done